logic/odmr_logic.py

```diff
@@ -208,15 +208,15 @@ def sv_get_fits(self, val):
     def _initialize_odmr_plots(self):
         """ Initializing the ODMR plots (line and matrix). """
 
-
-
         final_freq_list = []
         for mw_start, mw_stop, mw_step in zip(self.mw_starts, self.mw_stops, self.mw_steps):
             freqs = np.arange(mw_start, mw_stop + mw_step, mw_step)
             final_freq_list.extend(freqs)
             self.frequency_lists.append(freqs)
 
-        if self.final_freq_list:
+        if type(self.final_freq_list) == list:
+            self.final_freq_list = np.array(final_freq_list)
+        if self.final_freq_list.all():
```

Comment:
Can you please specify what the purpose of this if statement is? I think it will give `True` in almost all use cases as `False` is obtained only if there is a zero in the np.array. It also yields `True` when the array is empty.